### PR TITLE
Rebalance time of parallel stages for pre-merge CI

### PR DESCRIPTION
**Pull request by pxLi**
Signed-off-by: Peixin Li <pxli@nyu.edu>

try rebalance parallel stages of premerge CI.

previously premege CI parallel stages' durations,
CI1  ~2h10m
CI2  ~1h30m

currently total ~1h50m.

Also include mvn retry update to github action mvn checks

### CODE REVIEW

**Comment by pxLi** [comment]
build

**Comment by pxLi** [comment]
build

**Comment by pxLi** [comment]
build

**Comment by pxLi** [comment]
build

**Comment by pxLi** [comment]
made multiple run, as we fixed a lot of memory leak cases we could also clean up some legacy OOM workaround

**Comment by pxLi** [comment]
build

File: jenkins/spark-premerge-build.sh [hunk 1]
```diff
@@ -130,15 +128,17 @@ ci_2() {
...
+    export TEST_PARALLEL=5
+    ./integration_tests/run_pyspark_from_build.sh
+    # enable avro test separately
     INCLUDE_SPARK_AVRO_JAR=true TEST='avro_test.py' ./integration_tests/run_pyspark_from_build.sh
     # export 'LC_ALL' to set locale with UTF-8 so regular expressions are enabled
     LC_ALL="en_US.UTF-8" TEST="regexp_test.py" ./integration_tests/run_pyspark_from_build.sh
+
+    # put some mvn tests here to balance durations of parallel stages
+    echo "Run spark320 mvn test..."
+    env -u SPARK_HOME $MVN_CMD -U -B $MVN_URM_MIRROR -Dbuildver=320 clean test $MVN_BUILD_ARGS \
```
Comment (jlowe) [inline jenkins/spark-premerge-build.sh:140]: We shade our jar, so normally we need to run at least the `package` phase to properly execute tests after a clean.

**Comment by pxLi** [comment]
build

File: jenkins/spark-premerge-build.sh (hunk 1 above)
Comment (pxLi) [reply jenkins/spark-premerge-build.sh]: got it, updated

**Review by jlowe — APPROVED** [review]

**Pull request merged**